src/fights.ts

```diff
@@ -1400,39 +1402,89 @@ const freeFightSources = [
   new FreeFight(
     () => {
       if (!have($item`closed-circuit pay phone`)) return false;
+      // Check if we have or can get Shadow Affinity
       if (have($effect`Shadow Affinity`)) return true;
-      if (get("_shadowAffinityToday")) return false;
+      if (!get("_shadowAffinityToday") && !ClosedCircuitPayphone.rufusTarget()) return true;
 
-      if (!ClosedCircuitPayphone.rufusTarget()) return true;
```

Comment:
The logic here cannot be reduced to the one that you've written. If `get("_shadowAffinityToday")` is `true`, we want to return `false` immediately.

In your combined case, `if (!get("_shadowAffinityToday") && !ClosedCircuitPayphone.rufusTarget())` does not return (`false`) when `get("_shadowAffinityToday")` is `true`, which could have us entering the `FreeFight` without the ability to obtain Shadow Affinity

Reply:
That's intentional on my part. Even if we can't get shadow affinity, if we're able to force an NC we can complete a quest for free.